### PR TITLE
Apply low-pass filter to RTF display

### PR DESCRIPTION
**Pull request by azeey**
# 🎉 New feature

Follow-up from https://github.com/gazebosim/gz-sim/pull/1942

## Summary
This adds a more aggressive low-pass filter than what was in gz-sim such that the RTF display is more stable and reflects the RTF of a longer window of time.

## Test it
Run `gz sim camera_sensor.sdf`. The RTF will display is much more stable than before. After a while, remove the image display and watch the RTF increase.

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by azeey** [comment]
@iche033 I've made the filter less aggressive than what I had https://github.com/gazebosim/gz-sim/pull/1942 so that it is responsive to changes like the image display being removed while still providing a stable RTF reading.

File: src/plugins/world_stats/WorldStats.cc [hunk 1]
```diff
@@ -57,6 +57,9 @@ namespace plugins
 
     /// \brief Holds iterations
     public: QString iterations;
+
+    public: std::optional<double> simTimeDelayed;
+    public: std::optional<double> realTimeDelayed;
```
Comment (iche033) [inline src/plugins/world_stats/WorldStats.cc:62]: add doxy comments?

**Review by iche033 — APPROVED** [review]
nice, looks good to me. Just minor comment about doc

File: src/plugins/world_stats/WorldStats.cc (hunk 1 above)
Comment (azeey) [reply src/plugins/world_stats/WorldStats.cc:67]: Done in https://github.com/gazebosim/gz-gui/pull/529/commits/09fd320ae80869c9cbb2948fd1f4382b5229cbb6

**Pull request merged**